flocker/node/agents/blockdevice.py

```diff
@@ -340,3 +504,113 @@ def get_device_path(self, blockdevice_id):
         )
         # May be None if the file hasn't been used for a loop device.
         return _device_for_path(volume_path)
+
+
+def _manifestation_from_volume(volume):
+    """
+    :param BlockDeviceVolume volume: The block device which has the
+        manifestation of a dataset.
+    :returns: A primary ``Manifestation`` of a ``Dataset`` with the same id as
+        the supplied ``BlockDeviceVolume``.
+    """
+    dataset = Dataset(
+        dataset_id=volume.dataset_id,
+        maximum_size=volume.size,
+    )
+    return Manifestation(dataset=dataset, primary=True)
+
+
+@implementer(IDeployer)
+class BlockDeviceDeployer(PRecord):
+    """
+    An ``IDeployer`` that operates on ``IBlockDeviceAPI`` providers.
+
+    :param unicode hostname: The IP address of the node that has this deployer.
+    :param IBlockDeviceAPI block_device_api: The block device API that will be
+        called upon to perform block device operations.
+    :ivar FilePath mountroot: The directory where block devices will be
+        mounted.
+    """
+    hostname = field(type=unicode, mandatory=True)
+    block_device_api = field(mandatory=True)
+    mountroot = field(type=FilePath, initial=FilePath(b"/flocker"))
```

Comment:
Not sure about this. We want the default mountroot to be /flocker but to be overridable for tests.